_docs/tasks/telemetry/fluentd.md

```diff
@@ -0,0 +1,405 @@
+            value: http://elasticsearch:9200
+        ports:
+        - containerPort: 5601
+          name: ui
+          protocol: TCP
+---
+```
+
+Create the resources:
+
+```bash
+kubectl apply -f logging-stack.yaml
+```
+
+You should see the following:
+
+```
+namespace "logging" created
+service "elasticsearch" created
+deployment "elasticsearch" created
+service "fluentd-es" created
+deployment "fluentd-es" created
+configmap "fluentd-es-config" created
+service "kibana" created
+deployment "kibana" created
+```
+
+## Configure Istio
+
+Now that there is a running Fluentd daemon, configure Istio with a new
+log type, and send those logs to the listening daemon. Create a new
+YAML file to hold configuration for the log stream that
+Istio will generate and collect automatically.
+
+Save the following as `fluentd-istio.yaml`:
+
+```
+# Configuration for logentry instances
+apiVersion: "config.istio.io/v1alpha2"
+kind: logentry
+metadata:
+  name: newlog
+  namespace: istio-system
+spec:
+  severity: '"info"'
+  timestamp: request.time
+  variables:
+    source: source.labels["app"] | source.service | "unknown"
+    user: source.user | "unknown"
+    destination: destination.labels["app"] | destination.service | "unknown"
+    responseCode: response.code | 0
+    responseSize: response.size | 0
+    latency: response.duration | "0ms"
+  monitored_resource_type: '"UNSPECIFIED"'
+---
+# Configuration for a fluentd handler
+apiVersion: "config.istio.io/v1alpha2"
+kind: fluentd
+metadata:
+  name: handler
+  namespace: istio-system
+spec:
+  address: "fluentd-es.logging:24224"
+---
+# Rule to send logentry instances to the fluentd handler
+apiVersion: "config.istio.io/v1alpha2"
+kind: rule
+metadata:
+  name: newlogtofluentd
+  namespace: istio-system
+spec:
+  match: "true" # match for all requests
+  actions:
+   - handler: handler.fluentd
+     instances:
+     - newlog.logentry
+---
+```
+
+Create the resources:
+
+```bash
+istioctl create -f fluentd-istio.yaml
+```
+
+The expected output is similar to:
+```
+Created config logentry/istio-system/newlog at revision 22374
+Created config fluentd/istio-system/handler at revision 22375
+Created config rule/istio-system/newlogtofluentd at revision 22376
+```
+
+Notice that the `address: "fluentd-es.logging:24224"` line in the
+handler config is pointing to the Fluentd daemon we setup in the
+example stack.
+
+## View the new logs
+
+1. Send traffic to the sample application.
+
+   For the
+   [BookInfo](/docs/guides/bookinfo.html#determining-the-ingress-ip-and-port)
+   sample, visit `http://$GATEWAY_URL/productpage` in your web browser
+   or issue the following command:
+
+   ```bash
+   curl http://$GATEWAY_URL/productpage
+   ```
+
+1. In a Kubernetes environment, setup port-forwarding for Kibana by
+   executing the following command:
+
+   ```bash
+   kubectl -n logging port-forward $(kubectl -n logging get pod -l app=kibana -o jsonpath='{.items[0].metadata.name}') 5601:5601 &
```

Comment:
i think in other Tasks, when we've started such a process, we add that to the set of cleanup steps too.